### ISSUE TITLE
call connection from android to web sometimes fail on HPB

### ISSUE THREAD

**Issue by mahibi** [nextcloud/talk-android#1889]
## Steps to reproduce
1. start a call from android
2. accept on webbrowser

### Expected behaviour
call connection is successful

### Actual behaviour
sometimes it works, but sometimes this behaviour occurs:
- web can hear android
- android can't hear web, no participant is shown on android

debugging results:
no offer from browser is received.
(Because of this the MediaStreamEvent at https://github.com/nextcloud/talk-android/blob/master/app/src/main/java/com/nextcloud/talk/webrtc/PeerConnectionWrapper.java#L349 is never thrown.)

####  android logs for a SUCCESSFUL call:
[success.txt](https://github.com/nextcloud/talk-android/files/8387937/success.txt)

#### android logs for a FAILED call:

[fail.txt](https://github.com/nextcloud/talk-android/files/8387940/fail.txt)

server logs:

```
Mar 30 16:20:59 sermo nextcloud-spreed-signaling[2709738]: hub.go:1723: Session NIygxjp... is not in the same call as session jFV77E..., not requesting offer
Mar 30 16:21:00 sermo nextcloud-spreed-signaling[2709738]: room.go:575: Session jFV77E... joined call 3853979093
```

comment about this from @danxuliu :
race condition ;-)
when the Android app requests the offer the browser hasn't finished joining the call yet, so the request fails
in the iOS app and in the WebUI this is addressed with brute force :-P
basically if a connection can not be established it is retried every 10 seconds until it succeeded or the other participant is no longer in the call

## Device information

**Talk version:** lowest version tested was v13 up to 3e941edc, same behaviour..

## Server information

tested on Sermo at 30.3.2022 (24.0.0 beta 1)

**Comment by Ivansss** [comment]
Talk iOS app only use that brute force mechanism when a peer connection failed:
https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/NCCallController.m#L1009-L1029 

We need to add it when requesting offers when we detect a new user has joined the call:
https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/NCCallController.m#L880

Remember that we need to check if the offer arrives in the meantime and stop the timer that requests new offers for that user:
https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/NCCallController.m#L678-L689
https://github.com/nextcloud/talk-ios/blob/master/NextcloudTalk/NCCallController.m#L697
